tests/cases/compiler/controlFlowLetVar.ts

```diff
@@ -0,0 +1,119 @@
+// @strictNullChecks: true
+
+declare let cond: boolean;
+
+function f1() {
+    let x;
+    if (cond) {
+        x = 1;
+    }
+    if (cond) {
+        x = "hello";
+    }
+    const y = x;

```

Comment:
if you have `if (cond) .. else` then `y: string | number`, right? Might be nice to have a case for that.